File: SimpleApiBackend/Domain/Entities/Property.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace SimpleApiBackend.Domain.Entities;

public class Property
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public ObjectId Id { get; set; }

    public string Name { get; set; }
    public string Address { get; set; }
    public int Price { get; set; }
    public string CodeInternal { get; set; }
    public int Year { get; set; }

    [BsonRepresentation(BsonType.ObjectId)] 
    public ObjectId OwnerId { get; set; }    

    public List<PropertyImage> Images { get; set; } = [];
    public List<PropertyTrace> Trace { get; set; } = [];
}

File: nextjs-frontend/src/types/property.ts
import { PropertyImage } from "./property-image";
import { PropertyOwner } from "./property-owner";
import { PropertyTrace } from "./property-trace";

export interface Property {
  id: number;
  name: string;
  address: string;
  price: number;
  code: string;
  year: number;
  ownerDetails: PropertyOwner;
  image: PropertyImage;
  trace: PropertyTrace[];
}
